app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java

```diff
@@ -381,10 +382,11 @@ public void onBindViewHolder(final RecyclerView.ViewHolder holder, final int pos
                             .setOnClickListener(new View.OnClickListener() {
                                 @Override
                                 public void onClick(View view) {
-                                    Intent i = new Intent(Intent.ACTION_VIEW);
+                                    Intent i = new Intent(zulipApp.getApplicationContext(), PhotoViewActivity.class);
                                     i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-                                    i.setData(Uri.parse(url));
+                                    i.putExtra(Intent.EXTRA_TEXT , url);
                                     zulipApp.startActivity(i);
+                                    ((Activity) context).overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
```

Comment:
@kunall17 I just tried to make it a lot simple as it is on web. That's why i didn't made much good transitions. And i guess we don't need such transitions just for displaying images.  As for now i just made title as the name of pic which is fetched in url and displayed it in toolbar's title.